app/helpers/newflow_form_helper.rb

```diff
@@ -42,29 +43,45 @@ def text_field(name:,
         desired_class_name = "#{desired_class_name} #{supplemental_class}"
       end
 
-      input = (
+      if numberonly
+        input = (
+        @f.number_field name,
```

Comment:
Instead of a `numberonly` option to the `text_field`, I would try to create a `number_field` function instead. But I'm guessing that you did consider it already, so just a suggestion.